src/main/java/mods/eln/transparentnode/FuelGenerator.kt

```diff
@@ -0,0 +1,291 @@
+            in 0.5..1.1 -> 1.000
+            in 1.1..1.2 -> 1.050
+            in 1.2..1.3 -> 1.100
+            in 1.3..1.4 -> 1.150
+            in 1.4..1.5 -> 1.5
+            else -> 1.5
+        }
+
+        val TankCapacityInBuckets = 2
+        val GeneratorBailOutVoltageRatio = 0.5
+        val MinimalLoadFractionOfNominalPower = 0.1
+        val VoltageStabilizationGracePeriod = 1.0
+    }
+
+    internal val tankEnergyCapacity = tankCapacityInSecondsAtNominalPower * nominalPower;
+
+    internal val main = obj?.getPart("main")
+    internal val switch = obj?.getPart("switch")
+    internal val cableRenderDescriptor = cable.render
+
+    init {
+        voltageLevelColor = VoltageLevelColor.fromCable(cable)
+    }
+
+    override fun setParent(item: net.minecraft.item.Item, damage: Int) {
+        super.setParent(item, damage)
+        Data.addEnergy(newItemStack())
+    }
+
+    fun draw(on: Boolean = false) {
+        main?.draw()
+        if (on) {
+            UtilsClient.drawLight(switch)
+        } else {
+            switch?.draw()
+        }
+    }
+
+    override fun use2DIcon() = true
+
+    override fun handleRenderType(item: ItemStack, type: IItemRenderer.ItemRenderType) = true
+
+    override fun shouldUseRenderHelper(
+            type: IItemRenderer.ItemRenderType, item: ItemStack,
+            helper: IItemRenderer.ItemRendererHelper) = type != IItemRenderer.ItemRenderType.INVENTORY
+
+    override fun renderItem(type: IItemRenderer.ItemRenderType, item: ItemStack, vararg data: Any) = when(type) {
+        IItemRenderer.ItemRenderType.INVENTORY -> super.renderItem(type, item, *data)
+        else -> {
+            objItemScale(obj)
+            GL11.glPushMatrix()
+            Direction.ZP.glRotateXnRef()
+            GL11.glTranslatef(0f, -1f, 0f)
+            GL11.glScalef(0.6f, 0.6f, 0.6f)
+            draw()
+            GL11.glPopMatrix()
+        }
+    }
+
+    override fun addInformation(itemStack: ItemStack, entityPlayer: EntityPlayer,
+                                list: MutableList<String>, par4: Boolean) {
+        super.addInformation(itemStack, entityPlayer, list, par4)
+
+        list.add(tr("Produces electricity using fuel."))
+        list.add("  " + tr("Nominal voltage: %1$ V", Utils.plotValue(cable.electricalNominalVoltage)))
+        list.add("  " + tr("Nominal power: %1$ W", Utils.plotValue(nominalPower)))
+    }
+}
+
+class FuelGeneratorElement(transparentNode: TransparentNode, descriptor: TransparentNodeDescriptor):
+        TransparentNodeElement(transparentNode, descriptor) {
+    internal var positiveLoad = NbtElectricalLoad("positiveLoad")
+    internal var powerSource = PowerSource("powerSource", positiveLoad)
+    internal var slowProcess = FuelGeneratorSlowProcess(this)
+    internal var descriptor = descriptor as FuelGeneratorDescriptor
+    internal val fuel = FluidRegistry.getFluid("fuel") ?: FluidRegistry.getFluid("lava")
+    internal var tankLevel = 0.0
+    internal var on = false
```

Comment:
I added a custom property delegate that can be used to automatically publish properties when set: https://github.com/Electrical-Age/ElectricalAge/blob/feature/fuel-generator/src/main/java/mods/eln/node/Publishable.kt